native/swift/Sources/wordpress-api/WordPressLoginClient.swift

```diff
@@ -52,20 +53,42 @@ public final class WordPressLoginClient {
         forSite proposedSiteUrl: String,
         application: Application
     ) async throws -> URL {
-        createApplicationPasswordAuthenticationUrl(
-            loginUrl: try await findLoginUrl(forSite: proposedSiteUrl),
+        try await client.apiDiscovery(siteUrl: proposedSiteUrl).loginURL(for: application)
+    }
+
+    /// Convert the callback URL into a set of authentication credentials
+    ///
+    public func credentials(from callbackUrl: URL) throws -> WpApiApplicationPasswordDetails {
+        try extractLoginDetailsFromUrl(url: callbackUrl.absoluteString)
+    }
+}
+
+public extension AutoDiscoveryAttemptSuccess {
+
+    func loginURL(for application: Application) async throws -> URL {
+        let loginUrl = try apiDetails.applicationPasswordAuthenticationUrl
+        return createApplicationPasswordAuthenticationUrl(
+            loginUrl: loginUrl,
             appName: application.name,
             appId: application.id,
             successUrl: application.successCallbackUrl,
             rejectUrl: application.failureCallbackUrl
         ).asURL()
     }
 
-    /// Convert the callback URL into a set of authentication credentials
-    ///
-    public func credentials(from callbackUrl: URL) throws -> WpApiApplicationPasswordDetails {
-        try extractLoginDetailsFromUrl(url: callbackUrl.absoluteString)
+}
+
+public extension WpApiDetails {
+
+    var applicationPasswordAuthenticationUrl: ParsedUrl {
```

Comment:
We definitely don't want to crash for this case – there are many sites that don't support application passwords.

IINM though, Rust should emit an error for this case, so we wouldn't hit this anyway (AFAIK)

Reply:
That is right. If I am not mistaken, we return an error if authentication url can't be found. However, the API still returns an `Option` for this, so it's sort of annoying.

I'll see if I can somehow expose this without it being `Option` when we get a successful discovery result.

Reply:
@jkmassel Good point. I have moved the function to `AutoDiscoveryAttemptSuccess` in https://github.com/Automattic/wordpress-rs/pull/658/commits/537c490fe364e772093359c0ae9e7ce3381c0e9a, which only exists if application password authentication is available.